### PR TITLE
Fixes #17728: List of nodes by version does not include windows nodes

### PR DESCRIPTION
**Pull request by ElaadF**
https://issues.rudder.io/issues/17728
dsc_agent

### CODE REVIEW

File: webapp/sources/rudder/rudder-web/src/main/webapp/javascript/rudder/homePage.js [hunk 1]
```diff
@@ -216,11 +216,6 @@ function doughnutChart (id,data,count,colors) {
                       break;
                   case 'nodeAgents':
                       query.query.where = [{
-                          objectType: "software"
-                        , attribute : "cn"
-                        , comparator: "eq"
-                        , value     : "rudder-agent"
```
Comment (ElaadF) [inline webapp/sources/rudder/rudder-web/src/main/webapp/javascript/rudder/homePage.js:222]: I'm not sure if this is important, if I follow the logic `Rudder agent (DSC)` whould be use when a version contains a windows node (`rudder-agent OR Rudder agent (DSC)`)

Comment (amousset) [reply webapp/sources/rudder/rudder-web/src/main/webapp/javascript/rudder/homePage.js:222]: the problem is that t will match any software in the same version, which is not that likely to happen but would be very surprising for the user.
Can't we use a regex to match both `rudder-agent` and `Rudder agent (DSC)` in the filter?

Comment (VinceMacBuche) [reply webapp/sources/rudder/rudder-web/src/main/webapp/javascript/rudder/homePage.js:222]: We should look for the version we have in the "management technology" in out inventory (that's an entry we control and now we set Rudder version within, it was not the case in the old days

**Comment by VinceMacBuche** [comment]
you can also do egexp on software name

something like ```rudder-agent|Rudder Agent (Dsc)``` may be tester

**Comment by VinceMacBuche** [comment]
Rudder.*|rudder-agent as regex works fine

**Comment by VinceMacBuche** [comment]
{"query":{"select":"nodeAndPolicyServer","composition":"And","where":[{"objectType":"software","attribute":"cn","comparator":"regex","value":"rudder-agent|Rudder.*"},{"objectType":"software","attribute":"softwareVersion","comparator":"regex","value":"6(.|~)2-1(.|~)22.*"}]}}

works on our lab

**Review by VinceMacBuche — APPROVED** [review]

**Pull request merged**